crates/bevy_ecs/src/schedule/stage.rs

```diff
@@ -45,7 +51,191 @@ impl_downcast!(Stage);
 ///
 /// The checker may report a system more times than the amount of constraints it would actually need
 /// to have unambiguous order with regards to a group of already-constrained systems.
-pub struct ReportExecutionOrderAmbiguities;
+///
+/// By default only a warning with the number of unresolved ambiguities detected will be reported per [`SystemStage`].
+/// This behavior can be changed by explicitly adding this resource using the following constructors:
+/// * [ReportExecutionOrderAmbiguities::off()] - Disables all messages reported by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::minimal()] - Displays only the number of unresolved ambiguities detected by the ambiguity checker.
+/// * [ReportExecutionOrderAmbiguities::verbose()] - Displays a full report of ambiguities detected by the ambiguity checker.
+///
+/// The ambiguity checker will ignore ambiguities within official Bevy crates.
+/// To ignore a custom crate, use [`ReportExecutionOrderAmbiguities::ignore`]
+/// with an list of crate names as an argument.
+///
+/// ## Example
+/// ```ignore
+/// # use bevy_app::App;
+/// # use bevy_ecs::schedule::ReportExecutionOrderAmbiguities;
+/// App::new()
+///    .insert_resource(ReportExecutionOrderAmbiguities::verbose().ignore(&["my_external_crate"]));
+/// ```
+pub struct ReportExecutionOrderAmbiguities {
+    pub level: AmbiguityReportLevel,
+    pub ignore_crates: Vec<String>,
+}
+
+const BEVY_CRATES: [&str; 110] = [
+    "bevy_app",
+    "bevy_asset",
+    "bevy_audio",
+    "bevy_core",
+    "bevy_derive",
+    "bevy_diagnostic",
+    "bevy_dylib",
+    "bevy_dynamic_plugin",
+    "bevy_ecs",
+    "bevy_gilrs",
+    "bevy_gltf",
+    "bevy_input",
+    "bevy_internal",
+    "bevy_log",
+    "bevy_macro_utils",
+    "bevy_math",
+    "bevy_pbr",
+    "bevy_reflect",
+    "bevy_render",
+    "bevy_scene",
+    "bevy_sprite",
+    "bevy_tasks",
+    "bevy_text",
+    "bevy_transform",
+    "bevy_ui",
+    "bevy_utils",
+    "bevy_wgpu",
+    "bevy_window",
+    "bevy_winit",
+    "bevy_ai",
+    "bevy_android",
+    "bevy_animation",
+    "bevy_async",
+    "bevy_base",
+    "bevy_bounds",
+    "bevy_box2d",
+    "bevy_bullet",
+    "bevy_camera",
+    "bevy_cli",
+    "bevy_collision",
+    "bevy_controller",
+    "bevy_csharp",
+    "bevy_curve",
+    "bevy_demo",
+    "bevy_editor",
+    "bevy_egui",
```

Comment:
this one already exists as a third party plugin, should it be ignored? https://crates.io/crates/bevy_egui

Reply:
This shouldn't be ignored IMO.

Reply:
It is listed on [reserved_crates](https://github.com/bevyengine/bevy-crate-reservations/blob/main/reserved_crates) tho. Should I remove it from `BEVY_CRATES` list?

Reply:
Hmm. Yes, given the current usage. I would also create a tiny PR to remove it from the list you got it from too.

Reply:
Done in latest commit

Reply:
Also created the PR on bevy-crate-reservations https://github.com/bevyengine/bevy-crate-reservations/pull/14